Separate removing writable dmg

jaymalasinha opened the pull request:
Taking three step approach to solving this:
* Separate the remove writeable dmg call
* puppetstrings option enables progress output on hdiutil
* Verify the checksum for the final image created
* check detach completes before compressing the dmg

These put together shouldresolve the hdiutil resource busy failure
during compress_dmg

Signed-off-by: Jaymala Sinha <jsinha@chef.io>

### Description

Briefly describe the new feature or fix here

--------------------------------------------------

#### Maintainers

Please ensure that you check for:

- [] If this change impacts git cache validity, it bumps the git cache
  serial number
- [] If this change impacts compatibility with omnibus-software, the
  corresponding change is reviewed and there is a release plan
- [] If this change impacts compatibility with the omnibus cookbook, the
  corresponding change is reviewed and there is a release plan

btm commented:
I'm concerned this might not fix the current issue because we still won't fail if `hdiutil detach "#{@device}"` fails, as it still isn't the last command, which is possibly the command that is failing.

Can we implement the `set -e` feature we were discussing in #omnibus internal slack, or at least add a `set -e` to the top of every shellout block that is more than one command?

jaymalasinha commented:
@btm I have added the output progress tracking option available with hdiutil to avoid the resource busy option.  I think inbuilt command option will work better in this case.  Also I have added the dmg verification step to catch the 0bytes error.

jaymalasinha commented:
https://buildkite.com/chef/inspec-inspec-master-omnibus-adhoc/builds/54 passed so this is ready to merge
